x-pack/legacy/plugins/ml/server/models/job_service/jobs.js

```diff
@@ -364,14 +364,17 @@ export function jobsProvider(callWithRequest) {
     ]);
 
     if (jobStats.jobs.length) {
-      const time = jobStats.jobs[0].data_counts.latest_record_timestamp;
+      const statsForJob = jobStats.jobs[0];
+      const time = statsForJob.data_counts.latest_record_timestamp;
       const progress = (time - start) / (end - start);
+      const isJobClosed = statsForJob.state === JOB_STATE.CLOSED;
       return {
         progress: (progress > 0 ? Math.round(progress * 100) : 0),
```

Comment:
not related to your changes, but maybe we should use `Math.floor` instead of `round`?

Reply:
The key parts are the checks for the datafeed having stopped and the job state moving to closed. The buttons in the UI currently rely just on `progress` going to 100, so maybe `floor` is avoided to prevent any rounding issues with the progress calculation using the `latest_record_timestamp` not being exactly the requested end `ms` value. @jgowdyelastic  would know.